warp10/src/main/java/io/warp10/standalone/InMemoryChunkSet.java

```diff
@@ -807,80 +807,80 @@ public long clean(long now) {
         }
       }
     }
-    
+
     Sensision.update(SensisionConstants.SENSISION_CLASS_CONTINUUM_STANDALONE_INMEMORY_GC_CHUNKS, Sensision.EMPTY_LABELS, dropped);
-    
+
     return droppedDatapoints;
   }
-  
+
   /**
    * Optimize all non current chunks by shrinking their buffers.
-   * 
+   *
    * @param now
    */
   long optimize(CapacityExtractorOutputStream out, long now, AtomicLong allocation) {
-    
+
     if (this.ephemeral) {
       return 0L;
     }
-    
+
     int currentChunk = chunk(now);
-    
+
     long reclaimed = 0L;
 
-    synchronized(this.chunks) {      
+    synchronized(this.chunks) {
       for (int i = 0; i < this.chunks.length; i++) {
         if (null == this.chunks[i] || i == currentChunk) {
           continue;
         }
         int size = this.chunks[i].size();
-        
+
         try {
           this.chunks[i].writeTo(out);
           int capacity = out.getCapacity();
-          
+
           if (capacity > size) {
             this.chunks[i].resize(size);
             allocation.addAndGet(size);
             reclaimed += (capacity - size);
-          }          
-        } catch (IOException ioe) {          
+          }
+        } catch (IOException ioe) {
         }
       }
     }
-    
+
     return reclaimed;
   }
-  
+
   /**
    * Delete datapoints whose timestamp if >= start and <= end
-   * 
+   *
    * @param start Lower timestamp to delete (inclusive)
    * @param end Upper timestamp to delete (inclusive)
    * @return
    */
   public long delete(long start, long end) {
     long count = 0L;
-    
+
     for (int i = 0; i < chunks.length; i++) {
-      if (!this.ephemeral && (chunkends[i] < start || chunkends[i] >= end + chunklen)) {
+      if (null == chunks[i] || !this.ephemeral && (chunkends[i] < start || chunkends[i] >= end + chunklen)) {
         continue;
       }
       synchronized(chunks[i]) {
         GTSEncoder encoder = new GTSEncoder();
         GTSDecoder decoder = chunks[i].getUnsafeDecoder(false);
         boolean deleted = false;
-        while (decoder.next()) {
-          if (decoder.getTimestamp() >= start && decoder.getTimestamp() <= end) {
-            deleted = true;
-            count++;
-            continue;
-          }
-          try {
+        try {
+          while (decoder.next()) {
+            if (decoder.getTimestamp() >= start && decoder.getTimestamp() <= end) {
+              deleted = true;
+              count++;
+              continue;
+            }
```

Comment:
Only `encoder.addValue` can throw an `IOException`, why did you enclose the whole loop?

Reply:
Ridiculous change on my side :-), rolling back